src/components/Pagination/index.jsx

```diff
@@ -21,17 +22,40 @@ const Pagination = ({
     siblingCount,
     pageSize,
   });
+  const { handlePaginationQueryParamsChange } = usePaginationQueryParams();
 
   if (pageNo === 0 || paginationRange.length < 2) {
     return null;
   }
 
   const onNext = () => {
-    !isLastPage && navigate(pageNo + 1);
+    if (isLastPage) return;
+
+    const nextPage = pageNo + 1;
+    if (navigate) {
+      navigate(nextPage);
+    } else {
+      handlePaginationQueryParamsChange(nextPage);
+    }
   };
 
   const onPrevious = () => {
-    !isFirstPage && navigate(pageNo - 1);
+    if (isFirstPage) return;
+
+    const previousPage = pageNo - 1;
+    if (navigate) {
+      navigate(previousPage);
+    } else {
+      handlePaginationQueryParamsChange(previousPage);
+    }
+  };
+
+  const onPaginationItemClick = page => {
+    if (navigate) {
+      navigate(page);
+    } else {
+      handlePaginationQueryParamsChange(page);
+    }
```

Comment:
Instead of doing this everywhere, do this immediately after running `usePaginationQueryParams` hook:
```js
if (!navigate) navigate = handlePaginationQueryParamsChange;
```

Now you can directly call `navigate(page)` from everywhere.